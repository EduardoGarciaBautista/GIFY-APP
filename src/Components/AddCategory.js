import React, { useState } from 'react';
import { PropTypes } from 'prop-types';

const AddCategory = ({ setcategories }) => {

    const [inputValue, setInputValue] = useState('');


    const handleOnChange = (e) => {
        setInputValue(e.target.value);

        console.log('EVENT HANDLE');
    }

    const handleSubmit = (event) => {
        event.preventDefault();

        console.log('Submitted');
        if (inputValue.length > 2) {
            setcategories(cats => [inputValue, ...cats]);

            console.log('setcategories executed');
            setInputValue('');
        }
    }

    return (

        <form onSubmit={handleSubmit}>
            <input
                type="text"
                placeholder="Buscar"
                value={inputValue}
                onChange={handleOnChange}
            />
        </form>

    )
}

AddCategory.propTypes = {
    setcategories: PropTypes.func.isRequired
}

export default AddCategory;



