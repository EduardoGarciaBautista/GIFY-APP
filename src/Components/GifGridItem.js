import React from 'react';
import PropTypes from 'prop-types'

export const GifGridItem = ({ title, id, url }) => {

    return (
        <div className="card animate__fadeIn" id="card">
            <div className="img-container">
                <img src={url} alt={title} />
            </div>
            <div className="footer-contet">
                <p>
                    {title}
                </p>
            </div>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}