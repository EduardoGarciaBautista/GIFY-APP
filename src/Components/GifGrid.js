import React from 'react';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types'


const GifGrid = ({ category }) => {

    const {loading, data: images} = useFetchGifs(category);

    return (
    
        <div className="card-grid">
         
         {loading ?
         <h1>Cargando...</h1>:
         <>
             {images.map(img => <GifGridItem key={img.id} {...img} />)}
         </>
         }
        </div>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;