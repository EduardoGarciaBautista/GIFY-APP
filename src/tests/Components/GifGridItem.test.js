import { shallow } from 'enzyme';
import React from 'react';
import '@testing-library/dom';
import { GifGridItem } from '../../Components/GifGridItem';



describe('Test for GifGridItem component', () => {

    const title = 'Title test';

    const url = 'https://url.img.com';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('should match component with snapshot', () => {
        
     

        expect(wrapper).toMatchSnapshot();
        
    });
    test('should have p wit title variable', () => {
        
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);

    });

    test('should have an image equal to props url', () => {
        
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);

        expect(img.prop('alt')).toBe(title)

    });

    test('should have animate__fadeIn class', () => {

        const div = wrapper.find('div');
        const className = 'animate__fadeIn';

        expect(div.hasClass(className)).toBe(true)
        
    });
    
})