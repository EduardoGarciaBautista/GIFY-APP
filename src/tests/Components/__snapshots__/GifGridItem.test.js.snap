// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for GifGridItem component should match component with snapshot 1`] = `
<div
  className="card animate__fadeIn"
  id="card"
>
  <div
    className="img-container"
  >
    <img
      alt="Title test"
      src="https://url.img.com"
    />
  </div>
  <div
    className="footer-contet"
  >
    <p>
      Title test
    </p>
  </div>
</div>
`;
